### PR TITLE
Add proper formatting section and add commit convention link

### PR DESCRIPTION
**Pull request by benoitdm-oslandia**
In order to cleanup the CONTRIBUTING.md of Qgis project, some file contents were remove and the `git.rst` of this repo updated.

Linked to https://github.com/qgis/QGIS/pull/58472

### CODE REVIEW

**Comment by benoitdm-oslandia** [comment]
@DelazJ the failure does not seem to be linked to my changes. Is there something I should do?

File: docs/developers_guide/git.rst [hunk 1]
```diff
@@ -172,6 +195,10 @@ Procedure
       git add path/to/your/files
       git commit -m "Add a comment describing your nice feature"
 
+    Without the `-m` option, a new editor window will open for you to write your commit message.
```
Comment (DelazJ) [inline docs/developers_guide/git.rst:198]:
```suggestion
    Without the ``-m`` option, a new editor window will open for you to write your commit message.
```

File: docs/developers_guide/git.rst [hunk 2]
```diff
@@ -172,6 +195,10 @@ Procedure
...
+
+    Commit description should follow `this recommendation <https://www.conventionalcommits.org/en/v1.0.0/#summary>`_.
```
[the remark this answers is not in the document]
Comment (benoitdm-oslandia) [reply docs/developers_guide/git.rst:200]: Not in QGIS project as I know! But it is more a suggestion than something mandatory :)

File: docs/developers_guide/git.rst (hunk 1 above)
Comment (benoitdm-oslandia) [reply docs/developers_guide/git.rst:198]: fixed

File: docs/developers_guide/git.rst [hunk 3]
```diff
@@ -162,6 +162,29 @@ Procedure
     Code your changes in your local disk with your usual IDE.
     Remember to write tests suite for your modifications, when appropriate.
 
+#. Proper formatting and spell check
+    Make sure your code is properly formatted and spelled by running the prepare commit script 
+    **before** issuing `git commit`.
```
[the remark this answers is not in the document]
Comment (benoitdm-oslandia) [reply docs/developers_guide/git.rst:167]: fixed

File: docs/developers_guide/git.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (DelazJ) [reply docs/developers_guide/git.rst]: Yes I understood and it's a nice idea.

**Pull request merged**